app/src/test/webapp/app/error/error.ctrl.test.js

```diff
@@ -178,4 +193,72 @@ describe("ErrorCtrl", function() {
         });
     });
   });
+
+  describe("success after calling resubmit", function() {
+    it("should display a success message when the message is resubmitted", function() {
+      spyOn(messageCenterService, 'add');
+
+      $controller("ErrorCtrl", {
+        $scope: $scope
+      });
+
+      $scope.message = resubmittableMessage;
+
+      $scope.resubmitMessage();
+      $scope.$apply();
+
+      expect(messageCenterService.add).toHaveBeenCalledWith('success', 'Resubmit successful!', { "timeout": 3000 });
+    });
+  });
+
+  describe("failure after calling resubmit", function() {
+    it("should display a failure message when a message isn't resubmittable", function(){
+
+      var messageWithoutResubmitHeader = {
+        id: 1
+      };
+
+      spyOn(messageCenterService, 'add');
+
+      $controller("ErrorCtrl", {
+        $scope: $scope
+      });
+
+      $scope.message = messageWithoutResubmitHeader;
+
+      $scope.resubmitMessage();
+
+      expect(messageCenterService.add).toHaveBeenCalledWith('warning', 'Message can not be resubmitted', {
+        timeout: 5000
+      });
+
+    });
+
+    beforeEach(function failedCallToResubmitBackend() {
+      msgSvc.resubmitMessage = function() {
+        return $q.when({
+          data: {
+            "status": "Failure",
+            "errorMessage": "Just failed"
+          }
+        });
+
+      };
+    });
```

Comment:
This beforeEach is confusing... it's applying to both tests in this block, but it is not at the top of the describe block, which is convention. From the descriptions of the tests, I think you just mean the test below to have a server problem, so I would just inline this with the below test